### PR TITLE
raise memorySize limit to 2048MB

### PR DESCRIPTION
**Pull request by TPei**
memorySize is limited to 512MB in wskdeploy, but not in OpenWhisk
#1033

### CODE REVIEW

**Comment by TPei** [comment]
Ah, hang on, general OpenWhisk memory limit still seems to be 512MB:https://github.com/apache/incubator-openwhisk/blob/master/docs/reference.md

**Comment by TPei** [comment]
I think the test failures are actually related to: https://github.com/apache/incubator-openwhisk/pull/4254, with things like `Invalid or missing runtime [swift:3.1.1] specified in manifest for the action [helloSwiftWithCode]`

File: utils/validation.go [hunk 1]
```diff
@@ -153,7 +153,7 @@ func LimitsMemoryValidation(memory *int) bool {
 	if memory == nil {
 		return true
 	}
-	if *memory < 128 || *memory > 512 {
+	if *memory < 128 || *memory > 2048 {
 		wskprint.PrintlnOpenWhiskWarning(wski18n.T(wski18n.ID_WARN_LIMITS_MEMORY_SIZE))
 		return false
```
Comment (TPei) [inline utils/validation.go]: alternatively we could keep the warning for >512MB, but still return true

File: utils/validation.go [hunk 2]
```diff
@@ -153,7 +153,7 @@ func LimitsMemoryValidation(memory *int) bool {
 	if memory == nil {
 		return true
 	}
-	if *memory < 128 || *memory > 512 {
+	if *memory < 128 || *memory > 2048 {
```
Comment (csantanapr) [inline utils/validation.go]: This is a quick fix for now, but I think the validation should be remove or do a validation against backend

**Comment by csantanapr** [comment]
After this is merge the wsk CLI repo need to update the dependency file to pick up this bugs

**Comment by rabbah** [comment]
I agree with Carlos. The limits should not be hardcoded in wskdeploy. This will likely need some systemic cleaning.

File: utils/validation.go (hunk 2 above)
Comment (mrutkows) [reply utils/validation.go]: agree

**Pull request merged**